srml/im-online/src/lib.rs

```diff
@@ -232,15 +253,15 @@ decl_module! {
 				&current_session,
 				&heartbeat.authority_index
 			);
-			let keys = Keys::get();
+			let keys = Keys::<T>::get();
 			let public = keys.get(heartbeat.authority_index as usize);
```

Comment:
@bkchr while you are thinking of a way for returning an encoded iterator you might also think of a way to return n'th element of a vector without decoding the whole thing :p

Reply:
That does not work, as we don't know where an element starts in the encoded data.

Reply:
ahh really? I thought we could simply find the offset in a collection that has a unified type. 
But now I see your point: even for simplest types like numbers the encoding of the same type with different values could have different lengths so nvmnd.